conda_forge_tick/url_transforms.py

```diff
@@ -48,23 +50,29 @@ def _v_munger(url):
 
 
 def _pypi_domain_munger(url):
-    for old_d, new_d in permutations(
-        ["https://pypi.io", "https://files.pythonhosted.org"],
-        2,
-    ):
+    for old_d, new_d in permutations(PYPI_URLS, 2):
         yield url.replace(old_d, new_d, 1)
 
 
 def _pypi_name_munger(url):
     bn = os.path.basename(url)
-    if (
-        url.startswith("https://pypi.io")
-        or url.startswith("https://files.pythonhosted.org")
-    ) and ("{{ version }}" in bn and "{{ name" not in bn):
-        yield os.path.join(os.path.dirname(url), "{{ name }}-{{ version }}.tar.gz")
+    dn = os.path.dirname(url)
+    dist_bn = os.path.basename(os.path.dirname(url))
+    is_sdist = url.endswith(".tar.gz")
+    is_pypi = any(url.startswith(pypi) for pypi in PYPI_URLS)
+    has_version = re.search(r"\{\{\s*version", bn)
 
+    # try the original url first, as a fallback (probably can't be removed?)
     yield url
 
+    if not (is_sdist and is_pypi and has_version):
+        return
+
+    # try static PEP625 name with PEP345 distribution name (_ not -)
+    yield os.path.join(dn, '%s-{{ version }}.tar.gz' % dist_bn.replace("-", "_"))
```

Comment:
I think the PEP normalization also transforms periods to underscores. Check with ruamel.yaml to confirm but in that case we will need one more yield here.

Reply:
[4face02](https://github.com/regro/cf-scripts/pull/2794/commits/4face02266bc5b2d2035c48ee5707bb08956e037)